crates/core/src/factories/core_fetch.rs

```diff
@@ -295,12 +296,19 @@ impl CoreFetch {
         outgoing_request_tx: Sender<OutgoingRequest>,
         outgoing_request_rx: Arc<tokio::sync::Mutex<Receiver<OutgoingRequest>>>,
         space_id: SpaceId,
-        transport: DynTransport,
+        transport: Weak<dyn Transport>,
```

Comment:
This feels like it may be a pretty common pattern. Do you think we should create an official `pub type WeakTransport = Weak<dyn Transport>;` in the api crate?

Reply:
Yes, that's a good idea. That gives me a place to write about this in the code for people to find

https://github.com/holochain/kitsune2/pull/180/commits/5996eca03de6b117bc6a3123c269b08bb8ae4423
https://github.com/holochain/kitsune2/pull/180/commits/27adb70458409647d72a28478b67eb7ddc30882c
https://github.com/holochain/kitsune2/pull/180/commits/77ec3588227677845cfa3f6d2c96dcae3c50048f

Reply:
woot :+1: